src/getNarrative.js

```diff
@@ -52,15 +52,11 @@ const getNarrative = async (req, res) => {
   try {
     utils.log(`Fetching narrative ${fetchURL} and streaming to client for parsing`);
     const response = await fetch(fetchURL);
-    if (response.status === 404) {
-      return res.status(404).send("The requested URL does not exist.");
-    } else if (!(response.status === 200 || response.status === 304)) {
-      throw new Error(`Failed to fetch ${fetchURL}: ${response.status} ${response.statusText}`);
-    }
-    res.set("Content-Type", "text/markdown");
-    return response.body.pipe(res);
```

Comment:
Like in `requestMainDataset` this final return needs to be wrapped in a new Promise. I was able to get `community/narratives/jameshadfield/scratch/example-tree-map-zoom` to load locally with: 
```javascript
const stream = response.body.pipe(res);
return new Promise((resolve, reject) => stream.on('close', () => resolve()));
```
I don't fully understand this either....I got to this by going through the [node-fetch docs on streams](https://github.com/node-fetch/node-fetch#streams)

Reply:
I avoided this by no longer streaming the response (using `.send` instead), as the server had already fetched the entire file before we started the stream. Do you have an approach you'd recommend / changes you'd recommend to my implementation?